eugl/mndwi.py

```diff
@@ -0,0 +1,381 @@
+        red_bname = "BAND-3"
+        green_bname = "BAND-2"
+        blue_bname = "BAND-1"
+
+    # search for the RGB bands
+    red_path = [f for f in paths if (product in f and red_bname in f)]
+    green_path = [f for f in paths if (product in f and green_bname in f)]
+    blue_path = [f for f in paths if (product in f and blue_bname in f)]
+
+    if not red_path:
+        raise Exception(f'could not find "{product}"{red_bname}')
+
+    if not green_path:
+        raise Exception(f'could not find "{product}"{green_bname}')
+
+    if not blue_path:
+        raise Exception(f'could not find "{product}"{blue_bname}')
+
+    return red_path[0], green_path[0], blue_path[0]
+
+
+def seadas_style_scaling(band_im, sfactor):
+    """
+    Perform a (NASA-OBPG) SeaDAS style scaling on a single band.
+    When applied to the red, green and blue bands this scaling
+    generates very impressive RGB's
+
+    Parameters
+    ----------
+    band_im : numpy.ndarray
+        reflectance image at either the  band [nRows, nCols]
+
+    sfactor : float or int
+        The multiplicative factor used to convert to reflectances
+
+    Returns
+    -------
+    scaled_im : numpy.ndarray [dtype = np.uint8]
+        Scaled and enhanced band with dimensions of [nRows, nCols].
+
+    Raises
+    ------
+        * Exception if len(rgb_ix) != 3
+
+    """
+
+    # specify coefficient used in transformation:
+    c1 = 0.091935692
+    c2 = 0.61788
+    c3 = 10.0
+    c4 = -0.015
+
+    scaled_im = c1 + c2 * np.arctan(c3 * (band_im / sfactor + c4))
+    scaled_im[scaled_im < 0] = 0
+    scaled_im[scaled_im > 1] = 1
+    return np.array(255 * scaled_im, order="C", dtype=np.uint8)
+
+
+def mndwi(wagl_h5_file, granule, out_fname):
+    """
+    Computes the mndwi for a given granule in a wagl h5 file.
+
+    Parameters
+    ----------
+    wagl_h5_file : str
+        wagl-water-atcor generated h5 file
+
+    granule : str
+        Group path of the granule within the h5 file
+
+    out_fname : str
+        Output filename of the h5 file
+    """
+
+    # specify the reflectance products to use in generating mndwi
+    # products = ["LMBADJ"]
+    products = ["LAMBERTIAN", "LMBSKYG", "LMBADJ"]
+
+    h5_fid = h5py.File(out_fname, "w")
+
+    # find the granule index in the wagl_h5_file
+    fid = h5py.File(wagl_h5_file, "r")
+    grp = fid[granule]
+    paths = find(grp, "IMAGE")
+
+    # get platform name
+    md = yaml.load(fid[granule + "/METADATA/CURRENT"][()], Loader=yaml.FullLoader)
+    platform_id = md["source_datasets"]["platform_id"]
+
+    # Get the 1st (lowerVal_arr) and 99th (upperVal_arr) percentikes
+    # for the each of the products. These arrays are very useful
+    # for creating MNDWI images, but this code will be removed once
+    # testing is complete.
+    lowerVal_arr = np.zeros([len(products)], order="C", dtype=np.float32)
+    upperVal_arr = np.zeros([len(products)], order="C", dtype=np.float32)
```

Comment:
Put this into a separate function call along with the other components related to the contrast enhancement.

Reply:
I've removed the lowerVal_arr and upperVal_arr as I can work around this when generating comparison figures